cpp_linter/rest_api/github_api.py

```diff
@@ -522,7 +572,178 @@ def _dismiss_stale_reviews(self, url: str):
+        """
+        repo_owner, repo_name = self.repo.split("/")
+        query = """
+        query {
+            repository(owner:"%s", name:"%s") {
+                pullRequest(number: %d) {
+                    id
+                    reviewThreads(last: 100) {
+                        nodes {
+                            id
+                            isResolved
+                            isCollapsed
+                            comments(first: 10) {
+                                nodes {
+                                    id
+                                    body
+                                    path
+                                    line
+                                    startLine
+                                    originalLine
+                                    originalStartLine
+                                    author {
+                                        login
+                                    }
+                                    pullRequestReview {
+                                        id
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        """ % (
+            repo_owner,
+            repo_name,
+            self.pull_request,
+        )
+        response = self.api_request(
+            url="https://api.github.com/graphql",
+            method="POST",
+            data=json.dumps(
+                {"query": query}
+            ),
+            strict=False
+        )
+        if response.status_code != 200:
+            logger.error(
+                "Could not get existing review comments: %d", response.status_code)
+            return
+        data = response.json()
+        found_threads = []
+        for thread in data["data"]["repository"]["pullRequest"]["reviewThreads"]["nodes"]:
+            for comment in thread["comments"]["nodes"]:
+                if (
+                    comment["id"]
+                    and (not no_dismissed or (thread["isResolved"] is False and thread["isCollapsed"] is False))
+                    and comment["author"]["login"] == "github-actions"
+                    and (comment["body"].strip().startswith("### clang-format") or comment["body"].strip().startswith("### clang-tidy"))
+                ):
+                    found_threads.append(thread)
+                    break
+        return found_threads
+
+    def _close_review_comment(self, thread_id: str, comment_id: str, delete: bool = True):
+        """Resolve or Delete an existing review comment.
+
+        :param thread_id: Thread ID for the conversation to close (only used when ``delete``==`False`).
+        :param comment_id: The comment ID of the comment within the requested thread to close (only used when ``delete``==`True`).
+        :param delete: `True` to delete the review comment, `False` to set it as resolved.
+        """
```

Comment:
Uses graphQL to either set a review comment conversation as resolved or to delete based on user options.